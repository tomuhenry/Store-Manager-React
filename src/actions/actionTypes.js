const actionTypes = {
    LOGIN_SUCCESS: "LOGIN_SUCCESS",
    LOGIN_FAIL: "LOGIN_FAIL"
}

export default actionTypes;