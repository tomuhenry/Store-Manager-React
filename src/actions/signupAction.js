import actionTypes from './actionTypes';
import axios from 'axios';
import { toast } from 'react-toastify';

const signupUrl =
  'https://tomuhenry-storemanagerapp.herokuapp.com/store-manager/api/v1/auth/signup';

export const fetchsignupSuccess = payload => ({
  type: actionTypes.SIGNUP_SUCCESS,
  payload: payload
});

export const fetchsignupFailure = errors => ({
  type: actionTypes.SIGNUP_FAIL,
  errors
});

const signupAction = userData => dispatch => {
  const token = localStorage.getItem('access_token');

  const headers = {
    'Content-Type': 'application/json',
    Authorization: `Bearer ${token}`
  };

  userData.rights = false;
  return axios
    .post(signupUrl, userData, {
      headers: headers
    })
    .then(response => {
      if (response.data.Message) {
        toast.success(`${response.data.Message}`);
        dispatch(fetchsignupSuccess(response.data));
      } else if (response.data.error) {
        toast.error(`${response.data.error}`);
        dispatch(fetchsignupFailure(response.data));
      } else if (response.data.Alert) {
        toast.error(`${response.data.Alert}`);
        dispatch(fetchsignupFailure(response.data));
      }
    })
    .catch(error => {
      if (error.response.data.Alert) {
        toast.error(error.response.data.Alert);
      }
      if (error.response.data.error) {
        toast.error(error.response.data.error);
      }
      dispatch(fetchsignupFailure(error.response));
    });
};

export default signupAction;
