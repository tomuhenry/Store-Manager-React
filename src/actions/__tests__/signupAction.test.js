import actionTypes from '../actionTypes';
import configureMockStore from 'redux-mock-store';
import mockAxios from 'axios';
import thunk from 'redux-thunk';
import signupAction, {
  fetchsignupFailure,
  fetchsignupSuccess
} from '../signupAction';
import 'babel-polyfill';

jest.mock('axios');

describe('fetchLogin', () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({});
  const mockUserData = {
    email: 'email@here.com',
    password: 'my password',
    name: 'myname'
  };

  afterEach = () => {
    mockAxios.reset();
  };

  it('dispatch actions for signup success', async () => {
    const response = {
        Message: 'signed up',
      };
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({ data: response })
    );
    await store.dispatch(signupAction(mockUserData));
    expect(store.getActions().length).toEqual(1);
  });

  it('dispatch actions for signup success', async () => {
    const response = {
        error: 'signed up',
      };
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({ data: response })
    );
    await store.dispatch(signupAction(mockUserData));
    expect(store.getActions().length).toEqual(2);
  });

  it('dispatch actions for signup success', async () => {
    const response = {
        Alert: 'signed up',
      };
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({ data: response })
    );
    await store.dispatch(signupAction(mockUserData));
    expect(store.getActions().length).toEqual(3);
  });

  it('should dispatch success action', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);
    const store = mockStore({});
    const mockUserData = {
      email: 'email@here.com',
      password: 'my password',
      name: 'myname'
    };

    mockAxios.post('MockURL', mockUserData);
    store.dispatch(fetchsignupSuccess(mockUserData));
    expect(store.getActions()).toEqual([
      {
        payload: mockUserData,
        type: actionTypes.SIGNUP_SUCCESS
      }
    ]);
    expect(store.getActions().length).toEqual(1);
  });

  it('should dispatch failing action', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);
    const store = mockStore({});
    const mockUserData = { password: 'my password' };

    mockAxios.post('MockURL', mockUserData);
    store.dispatch(fetchsignupFailure(mockUserData));
    expect(store.getActions()).toEqual([
      {
        errors: mockUserData,
        type: actionTypes.SIGNUP_FAIL
      }
    ]);
    expect(store.getActions().length).toEqual(1);
  });
});
