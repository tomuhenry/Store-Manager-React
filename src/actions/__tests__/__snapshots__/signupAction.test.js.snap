// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Actions dispatch actions for signup 1`] = `<signupAction />`;
